framework/web/Session.php

```diff
@@ -129,6 +131,7 @@ public function open()
         if ($this->getIsActive()) {
             Yii::info('Session started', __METHOD__);
             $this->updateFlashCounters();
+            register_shutdown_function([$this, 'close']);
```

Comment:
yep, should be moved back to `init()`. `close()` will be called only once and will only take action if a session is still open.